code/Control/UserDefinedFormController.php

```diff
@@ -556,6 +556,14 @@ public function process($data, $form)
             $session->set('userformssubmission'. $this->ID, $submittedForm->ID);
         }
 
+        // If set, redirect to custom thank you page
+        if ($this->data()->ConfirmationPageID) {
+            $confirmationPage = $this->data()->ConfirmationPage();
+            if ($confirmationPage && $confirmationPage->exists()) {
```

Comment:
```suggestion
        $confirmationPage = $this->data()->ConfirmationPage();
        if ($confirmationPage && $confirmationPage->exists()) {
```
This is a simpler way to do the condition - though not by any means necessary for merging.